Subject: [PATCH] add missing json annotations
---
Index: apis/gateway/v1beta1/apirule_types.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/apis/gateway/v1beta1/apirule_types.go b/apis/gateway/v1beta1/apirule_types.go
--- a/apis/gateway/v1beta1/apirule_types.go	(revision f5caf61c353a1387a0f59f748c76b985640fae10)
+++ b/apis/gateway/v1beta1/apirule_types.go	(revision 288b8b3ba7ff7cb154f11d016aa8779d6e61d708)
@@ -43,19 +43,19 @@
 	// +kubebuilder:validation:MinLength=3
 	// +kubebuilder:validation:MaxLength=256
 	// +kubebuilder:validation:Pattern=^([a-zA-Z0-9][a-zA-Z0-9-_]*\.)*[a-zA-Z0-9]*[a-zA-Z0-9-_]*[[a-zA-Z0-9]+$
-	Host *string `json:"host"`
+	Host *string `json:"host,omitempty"`
 	// Describes the service to expose.
 	// +optional
 	Service *Service `json:"service,omitempty"`
 	// Specifies the Istio Gateway to be used.
 	// +kubebuilder:validation:Pattern=`^[0-9a-z-_]+(\/[0-9a-z-_]+|(\.[0-9a-z-_]+)*)$`
-	Gateway *string `json:"gateway"`
+	Gateway *string `json:"gateway,omitempty"`
 	// Specifies CORS headers configuration that will be sent downstream
 	// +optional
 	CorsPolicy *CorsPolicy `json:"corsPolicy,omitempty"`
 	// Represents the array of Oathkeeper access rules to be applied.
 	// +kubebuilder:validation:MinItems=1
-	Rules []Rule `json:"rules"`
+	Rules []Rule `json:"rules,omitempty"`
 	// +optional
 	Timeout *Timeout `json:"timeout,omitempty"`
 }
@@ -90,7 +90,7 @@
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	Spec   APIRuleSpec   `json:"spec,omitempty"`
-	Status APIRuleStatus `json:"status,omitempty"`
+	Status APIRuleStatus `json:"status,omitempty,omitzero"`
 }
 
 // +kubebuilder:object:root=true
Index: apis/gateway/v2/apirule_types.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/apis/gateway/v2/apirule_types.go b/apis/gateway/v2/apirule_types.go
--- a/apis/gateway/v2/apirule_types.go	(revision f5caf61c353a1387a0f59f748c76b985640fae10)
+++ b/apis/gateway/v2/apirule_types.go	(revision 288b8b3ba7ff7cb154f11d016aa8779d6e61d708)
@@ -38,20 +38,20 @@
 	// Specifies the URLs of the exposed service.
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=1
-	Hosts []*Host `json:"hosts"`
+	Hosts []*Host `json:"hosts,omitempty"`
 	// Describes the service to expose.
 	// +optional
 	Service *Service `json:"service,omitempty"`
 	// Specifies the Istio Gateway to be used.
 	// +kubebuilder:validation:MaxLength=127
 	// +kubebuilder:validation:XValidation:rule=`self.matches('^[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?/([a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)$')`,message="Gateway must be in the namespace/name format"
-	Gateway *string `json:"gateway"`
+	Gateway *string `json:"gateway,omitempty"`
 	// Specifies CORS headers configuration that will be sent downstream
 	// +optional
 	CorsPolicy *CorsPolicy `json:"corsPolicy,omitempty"`
 	// Represents the array of Oathkeeper access rules to be applied.
 	// +kubebuilder:validation:MinItems=1
-	Rules []Rule `json:"rules"`
+	Rules []Rule `json:"rules,omitempty"`
 	// +optional
 	Timeout *Timeout `json:"timeout,omitempty"`
 }
@@ -88,7 +88,7 @@
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	Spec   APIRuleSpec   `json:"spec,omitempty"`
-	Status APIRuleStatus `json:"status,omitempty"`
+	Status APIRuleStatus `json:"status,omitempty,omitzero"`
 }
 
 // +kubebuilder:object:root=true
@@ -137,10 +137,10 @@
 	Service *Service `json:"service,omitempty"`
 	// Represents the list of allowed HTTP request methods available for the **spec.rules.path**.
 	// +kubebuilder:validation:MinItems=1
-	Methods []HttpMethod `json:"methods"`
+	Methods []HttpMethod `json:"methods,omitempty"`
 	// Disables authorization when set to true.
 	// +optional
-	NoAuth *bool `json:"noAuth"`
+	NoAuth *bool `json:"noAuth,omitempty"`
 	// Specifies the Istio JWT access strategy.
 	// +optional
 	Jwt *JwtConfig `json:"jwt,omitempty"`
Index: apis/gateway/v2alpha1/apirule_types.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/apis/gateway/v2alpha1/apirule_types.go b/apis/gateway/v2alpha1/apirule_types.go
--- a/apis/gateway/v2alpha1/apirule_types.go	(revision f5caf61c353a1387a0f59f748c76b985640fae10)
+++ b/apis/gateway/v2alpha1/apirule_types.go	(revision 288b8b3ba7ff7cb154f11d016aa8779d6e61d708)
@@ -37,20 +37,20 @@
 	// Specifies the URLs of the exposed service.
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=1
-	Hosts []*Host `json:"hosts"`
+	Hosts []*Host `json:"hosts,omitempty"`
 	// Describes the service to expose.
 	// +optional
 	Service *Service `json:"service,omitempty"`
 	// Specifies the Istio Gateway to be used.
 	// +kubebuilder:validation:MaxLength=127
 	// +kubebuilder:validation:XValidation:rule=`self.matches('^[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?/([a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)$')`,message="Gateway must be in the namespace/name format"
-	Gateway *string `json:"gateway"`
+	Gateway *string `json:"gateway,omitempty"`
 	// Specifies CORS headers configuration that will be sent downstream
 	// +optional
 	CorsPolicy *CorsPolicy `json:"corsPolicy,omitempty"`
 	// Represents the array of Oathkeeper access rules to be applied.
 	// +kubebuilder:validation:MinItems=1
-	Rules []Rule `json:"rules"`
+	Rules []Rule `json:"rules,omitempty"`
 	// +optional
 	Timeout *Timeout `json:"timeout,omitempty"`
 }
@@ -88,7 +88,7 @@
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	Spec   APIRuleSpec   `json:"spec,omitempty"`
-	Status APIRuleStatus `json:"status,omitempty"`
+	Status APIRuleStatus `json:"status,omitempty,omitzero"`
 }
 
 // +kubebuilder:object:root=true
@@ -137,7 +137,7 @@
 	Service *Service `json:"service,omitempty"`
 	// Represents the list of allowed HTTP request methods available for the **spec.rules.path**.
 	// +kubebuilder:validation:MinItems=1
-	Methods []HttpMethod `json:"methods"`
+	Methods []HttpMethod `json:"methods,omitempty"`
 	// Disables authorization when set to true.
 	// +optional
 	NoAuth *bool `json:"noAuth,omitempty"`
